python/tank/util/user_settings.py

```diff
@@ -36,101 +42,168 @@ def _init_singleton(self):
         """
         Singleton initialization.
         """
-        path = self._compute_config_location()
-        logger.debug("Reading user settings from %s" % path)
+        self._path = self._compute_config_location()
+        logger.debug("Reading user settings from %s" % self._path)
 
-        self._user_config = self._load_config(path)
+        self._user_config = self._load_config(self._path)
 
         # Log the default settings
-        logger.debug("Default site: %s" % (self.default_site or "<missing>",))
-        logger.debug("Default login: %s" % (self.default_login or "<missing>",))
+        logger.debug("Default site: %s", self._to_friendly_str(self.default_site))
+        logger.debug("Default login: %s", self._to_friendly_str(self.default_login))
 
         self._settings_proxy = self._get_settings_proxy()
         self._system_proxy = None
-        if self._settings_proxy:
-            logger.debug("Shotgun proxy (from settings): %s" % self._get_filtered_proxy(self._settings_proxy))
+        if self._settings_proxy is not None:
+            logger.debug("Shotgun proxy (from settings): %s", self._get_filtered_proxy(self._settings_proxy))
         else:
             self._system_proxy = self._get_system_proxy()
-            if self._system_proxy:
-                logger.debug("Shotgun proxy (from system): %s" % self._get_filtered_proxy(self._system_proxy))
+            if self._system_proxy is not None:
+                logger.debug("Shotgun proxy (from system): %s", self._get_filtered_proxy(self._system_proxy))
             else:
                 logger.debug("Shotgun proxy: <missing>")
 
         proxy = self._get_filtered_proxy(self.app_store_proxy)
-        if self.is_app_store_proxy_set():
-            logger.debug("App Store proxy: %s" % (proxy or "<empty>",))
-        else:
-            logger.debug("App Store proxy: <missing>")
+        logger.debug("App Store proxy: %s", self._to_friendly_str(proxy))
+
+    @property
+    def path(self):
+        """
+        Retrieves the path to the current user settings file.
+        """
+        return self._path
```

Comment:
What does the path mean? A proxy setting can already come from a series of overrides - i am worrying that we are locking outselves in by adding a `path` like this - i think we talked about this before - what if we want to read a series of files, each with overrides? Or store things in a database? Part of the benefit with the `UserSettings` object was to provide an abstraction around a potentially complex backend.